task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from pyrsistent import field
+from pyrsistent import PMap
+from pyrsistent import PVector
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    node_selector = field(type=PMap)
+    containers = field(type=PVector)
+    # Hardcoded for the time being
+    restart_policy = "Never"
+    # By default, the retrying executor retries 3 times. This task option
+    # overrides the executor setting.
+    retries = field(type=int,
+                    factory=int,
+                    mandatory=False,
+                    invariant=lambda r: (r >= 0, 'retries >= 0'))
+    volumes = field(type=PVector)
+
+    @property
+    def pod_name(self) -> str:
+        return self.name
+
+    def set_pod_name(self, pod_name: str):
+        try:
+            name, uuid = pod_name.rsplit('.', maxsplit=1)
+        except ValueError:
+            raise ValueError(f'Invalid format for pod_name {pod_name}')
+
+        if len(pod_name) > 253:
```

Comment:
validation on composite fields like this belong in `__invariant__` method, see here: https://github.com/Yelp/task_processing/blob/cf843658d2af1122e7aeb279200a75aa60c785f5/task_processing/plugins/mesos/task_config.py#L48

Reply:
Aah thank you! Moving validation to __invariant__.